gitpod-io/gitpod
# [server] Use image-builder from workspace cluster (feature flag: "movedImageBuilder")

## Description

This is part II of a translation from [Chris' PR](https://github.com/gitpod-io/gitpod/pull/9337). To be able to move forward with this, it got split into two:
 1. The deprecation of the old log retrieval mechanism in [this PR](https://github.com/gitpod-io/gitpod/pull/11026) (feature flag: "deprecateOldImageLogsMechanism")
 2. Make `server` talk to `ws-manager` for image builds (this PR) (feature flag: "movedImageBuilder")
 
The idea is to get both merged, and be able to test and iterate independently.

/cc @kylos101 

**Note 1**: All formatting should be in commit 1, so it's sufficient to review commit 2 alone.

**Note 2**: Commit 3 + 4 do not strictly belong into this PR but I left them here because they ease testing. :lotus_position: 

## Related Issue(s)

Fixes #9248

Context: #9337

## How to test
 - [preview env](https://gpl-imgbldr.preview.gitpod-dev.com/workspaces): create a team with a name containing `imgbldr` and project (e.g. on this [test repo](https://github.com/geropl/gitpod-test-repo/tree/gpl/docker-build-2mins)) (the `imgbldr` triggers feature flags  `movedImageBuilder` and `deprecateOldImageLogsMechanism`, cmp. [here](https://app.configcat.com/08da1258-64fb-4a8e-8a1e-51de773884f6/08da1258-6541-4fc7-8b61-c8b47f82f3a0/08da1258-6512-4ec0-80a3-3f6aa301f853))
 - push a commit, for instance to [this branch](https://github.com/geropl/gitpod-test-repo/tree/gpl/docker-build-2mins)
 - [open a workspace on that branch](https://gpl-imgbldr.preview.gitpod-dev.com/#https://github.com/geropl/gitpod-test-repo/tree/gpl/docker-build-2mins)
 - observe that workspace start and the streaming of image build logs works as expected
 - check `server` logs for the line `image-builder in workspace cluster` to validate you actually used the new mechanism

## Release Notes

```release-note
NONE
```

## Documentation


## Werft options:
